1.16.1/SNP_DirectIP/src/DirectIP/Settings.cpp

```diff
@@ -0,0 +1,26 @@
+#include "Settings.h"
+
+#include <iostream>
+#include <fstream>
+
+#include "SettingsFile.h"
+#include "SettingsDialog.h"
+#include "SettingsEnvironment.h"
+
+Settings* Settings::getSettings() {
+	SettingsEnvironment* envSettings = new SettingsEnvironment();
+	if (envSettings->isOk()) {
+		return envSettings;
+	}
+	else {
+		delete envSettings;
+	}
+
+	std::ifstream settingsFile("directip.conf", std::ifstream::in);
+	if (settingsFile.good()) {
+		return new SettingsFile(settingsFile);
+	}
+	else {
+		return new SettingsDialog();
+	}
+}
```

Comment:
Can you use `std::shared_ptr` for these? I'd like the project to stop relying on manual memory management.

Reply:
Yes, will update.